### PR TITLE
[RUN-2989] DevicePixelRatio initially wrong on Mac ARM

### PR DESCRIPTION
**Pull request by Domiii**
* https://linear.app/replay/issue/RUN-2989/devicepixelratio-initially-wrong-on-mac-arm#comment-6b6b2b82
* I tested it as explained [here](https://linear.app/replay/issue/RUN-2989/devicepixelratio-initially-wrong-on-mac-arm#comment-3fcef722).

### CODE REVIEW

File: base/BUILD.gn [hunk 1]
```diff
@@ -624,6 +624,8 @@ mixed_component("base") {
     "record_replay.cc",
     "record_replay.h",
     "record_replay_ordered_atomic.h",
+    "record_replay_render_interface.h",
```
Comment (Domiii) [inline base/BUILD.gn:627]: This code helps blink access our render code. Move it to its own file.

File: third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc [hunk 2]
```diff
@@ -94,13 +95,13 @@ extern "C" char* V8RecordReplayReadAssetFileContents(const char* aPath, size_t*
 static const char REPLAY_CDT_PAUSE_OBJECT_GROUP[] =
     "REPLAY_CDT_PAUSE_OBJECT_GROUP";
 
-static bool IsGReplayScriptEnabledWhenRecording() {
+static bool IsCommandHandlingEnabledWhenRecording() {
```
Comment (Domiii) [inline third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc:98]: Sneaky: Rename another `GReplayScript` reference, while we are at it.

File: components/viz/service/display/software_output_device.h [hunk 3]
```diff
@@ -83,6 +83,11 @@ class VIZ_SERVICE_EXPORT SoftwareOutputDevice {
   // platform's proposed surface size.
   virtual bool SupportsOverridePlatformSize() const;
 
+  // [RUN-2989] Allow Replay commmand handlers to read the viewport size.
+  const gfx::Size& ReplayViewportPixelSize() const {
+    return viewport_pixel_size_;
```
Comment (toshok) [inline components/viz/service/display/software_output_device.h:88]: mind making this function return `gfx::Size*` as well and returning `&viewport_pixel_size_` here (and removing the  `&` from `GetCurrentViewportPixelSizeImpl`)?

the current formulation of things makes `GetCurrentViewportPixelSizeImpl` look like it's opening us up to a use-after-free.

**Review by toshok — APPROVED** [review]
request inline.  vaguely weirded out by this change, since it assumes that at all points in the execution after that initial gray area, our `Graphics_getDevicePixelRatio` in js will yield the same result as `window.devicePixelRatio`.

can we not change the implementation of `window.devicePixelRatio` to do what we want instead? ifdef'ed/commented so that if/when its implementation changes due to rebasing or something else, we'll know.

**Comment by Domiii** [comment]
@toshok 
> request inline. vaguely weirded out by this change, since it assumes that at all points in the execution after that initial gray area, our `Graphics_getDevicePixelRatio` in js will yield the same result as `window.devicePixelRatio`.
> 
> can we not change the implementation of `window.devicePixelRatio` to do what we want instead? ifdef'ed/commented so that if/when its implementation changes due to rebasing or something else, we'll know.

I was considering that, but I considered two (but really just one) guidelines:
1. We generally try to avoid changing Chromium behavior.
2. Brian also generally recommends our changes to be as unintrusive as possible.

Maybe a better fix would be to rename the command, so we clearly distinguish this command from `devicePixelRatio`. But changing the protocol feels like an uphill battle involving a lot of players, which is why I opted to lazy out on that 👀

**Pull request merged**